### PR TITLE
allow offers request on orderbook subscribe when disconnected

### PR DESCRIPTION
**Pull request by shekenahglory**
When the remote is offline and an orderbook is created with XRP as the `gets` currency, the orderbook never gets properly subscribed because requestOffers doesn't allow a request to go to the queue when remote is disconnected.  Its not a problem for non-native currencies because requestTransferRate waits for the remote to connect.

### CODE REVIEW

**Comment by FrRichard** [comment]
is it related to https://github.com/ripple/ripple-lib/issues/573 ?

**Comment by shekenahglory** [comment]
Seems likely - if you are subscribing before connecting

**Comment by FrRichard** [comment]
hmm this is weird because I always remote.connect() before .. I'll try with those changes and see

**Comment by FrRichard** [comment]
this bug didn't appear with 0.12.6

**Comment by shekenahglory** [comment]
remote.connect() is async, so it may be that you initiated the connection but hasnt been completed yet - unless you are specifically waiting until you get a connection response

**Comment by FrRichard** [comment]
got it thanks

**Comment by clark800** [comment]
Is it possible to add a unit test to prevent regressions? Could you add a comment explaining what `internal` is and how it is used?

**Comment by shekenahglory** [comment]
Yeah I can add that, I'm not entirely sure that the bit about ignoring requests while its not connected is ideal though...it doesn't seem like expected behavior does it?

**Comment by shekenahglory** [comment]
It seems like we would want to ignore subsequent requests if one is already queued up, not ignore all

**Comment by clark800** [comment]
I think the Request class will call this method again when it reconnects, until it gets a success or error response or times out.

**Comment by shekenahglory** [comment]
As it is now, if requestOffers is called when its disconnected, a request object never gets created - it just returns undefined without even executing the callback

**Comment by shekenahglory** [comment]
Actually isnt this always an internal call and therefor a private method? there is another method `getOffers` that is public

**Comment by shekenahglory** [comment]
@darkdarkdragon I also just discovered that autobridged books don't load because `syncronized` is never set to `true`: https://github.com/ripple/ripple-lib/blob/develop/src/core/orderbook.js#L358 I thinking this supposed to be `synced` here and under `computeAutobridgedOffersWrapper` also?

**Comment by darkdarkdragon** [comment]
@shekenahglory You're right, it must be `_synced`. Missed that one on merge :confused:

File: src/core/orderbook.js [hunk 1]
```diff
@@ -333,16 +336,18 @@ OrderBook.prototype.destroy = function() {
...
 OrderBook.prototype.requestOffers = function(callback = function() {},
   internal = false) {
   const self = this;
 
-  if (!this._remote.isConnected()) {
+  if (!this._remote.isConnected() && !internal) {
     // do not make request if not online.
     // that requests will be queued and
     // eventually all of them will fire back
+    callback(new Error('remote is offline'));
```
Comment (clark800) [inline src/core/orderbook.js]: All errors passed to callbacks should be instances of RippleError (which means they can inherit from RippleError).

**Comment by clark800** [comment]
LGTM after squash

**Comment by shekenahglory** [comment]
Rebased and squashed, but now I get seemingly unrelated test failures?

`ReferenceError: UInt160 is not defined
      at Remote.requestGatewayBalances (/home/ubuntu/ripple-lib/src/core/remote.js:1:14)`

**Comment by clark800** [comment]
Yes, that was a merge failure that happened yesterday. I'll fix it soon, thanks

**Comment by clark800** [comment]
I fixed the bug and tried rebuilding on CircleCI, but it failed again. Can you try rebasing?

**Pull request merged**